system/modules/core/drivers/DC_Folder.php

```diff
@@ -2393,22 +2393,39 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=f
 			$return .= "\n  " . '<li class="tl_file click2edit toggle_select" onmouseover="Theme.hoverDiv(this,1)" onmouseout="Theme.hoverDiv(this,0)"><div class="tl_left" style="padding-left:'.($intMargin + $intSpacing).'px">';
 
 			// Generate the thumbnail
-			if ($objFile->isImage && $objFile->height > 0)
+			if ($objFile->isImage && $objFile->viewHeight > 0)
 			{
-				$popupWidth = ($objFile->width > 600) ? ($objFile->width + 61) : 661;
-				$popupHeight = ($objFile->height + 200);
-				$thumbnail .= ' <span class="tl_gray">('.$this->getReadableSize($objFile->filesize).', '.$objFile->width.'x'.$objFile->height.' px)</span>';
+				if ($objFile->width && $objFile->height)
+				{
+					$popupWidth = ($objFile->width > 600) ? ($objFile->width + 61) : 661;
+					$popupHeight = ($objFile->height + 210);
+				}
+				else
+				{
+					$popupWidth = 661;
+					$popupHeight = 625 / $objFile->viewWidth * $objFile->viewHeight + 210;
+				}
+
+				$thumbnail .= ' <span class="tl_gray">('.$this->getReadableSize($objFile->filesize);
+				if ($objFile->width && $objFile->height)
+				{
+					$thumbnail .= ', '.$objFile->width.'x'.$objFile->height.' px';
+				}
+				$thumbnail .= ')</span>';
 
 				if (\Config::get('thumbnails') && ($objFile->isSvgImage || $objFile->height <= \Config::get('gdMaxImgHeight') && $objFile->width <= \Config::get('gdMaxImgWidth')))
 				{
-					$_height = ($objFile->height < 50) ? $objFile->height : 50;
-					$_width = (($objFile->width * $_height / $objFile->height) > 400) ? 90 : '';
+					$_height = ($objFile->height && $objFile->height < 50) ? $objFile->height : 50;
 
-					$thumbnail .= '<br><img src="' . TL_FILES_URL . \Image::get($currentEncoded, $_width, $_height) . '" alt="" style="margin:0 0 2px -19px">';
+					$thumbnail .= '<br><img src="' . TL_FILES_URL . \Image::get($currentEncoded, 400, $_height, 'box') . '" alt="" style="margin:0 0 2px -19px">';
 				}
 			}
 			else
 			{
+				if ($objFile->isImage)
```

Comment:
That's probably because it follows the `$popupHeight = $objFile->height + 210` approach :)

Ok, I'm going to merge this now. Thanks for your patience.